core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/service/ClusterProxyService.java

```diff
@@ -225,8 +243,16 @@ private String internalAdminUrl(Stack stack, int port) {
         return String.format("https://%s:%d", gatewayIp, port);
     }
 
-    public String getProxyPath(String crn) {
-        return String.format("%s/proxy/%s/%s", clusterProxyConfiguration.getClusterProxyBasePath(), crn, CB_INTERNAL);
+    public String getProxyPath(String crn, String internalIdentity) {
+        ReadConfigResponse readConfigResponse = clusterProxyRegistrationClient.readConfig(crn);
```

Comment:
while I understand the motivation of checking this, but this could lead to extra load on CP side as well as adding delay to client building. I can see 2 way for improving this:
- store locally when we register this, then we don't have to call CP
- cache the response to avoid calls

what do you think?

Reply:
I think CP does only a few things, so I really hope it can read it very fast. But if I could choose between storing it or caching I would go with storing this. Please discuss in private.

Reply:
I asked CP guys and it is OK to call readConfig on every client creation